webapp/src/components/Quiz/Quiz.js

```diff
@@ -1,144 +1,254 @@
-import React, { useState, useRef } from 'react'
-import Countdown, { zeroPad } from 'react-countdown'
+import "./Quiz.css";
+import React, { useState, useRef, useEffect } from "react";
+import Countdown, { zeroPad } from "react-countdown";
+import Question from "./Question";
+import FinalResult from "../FinalResult/FinalResult";
+import CircularProgress from '@mui/material/CircularProgress';
+import { ReactComponent as StopwatchIcon } from "../../assets/stopwatch-solid.svg";
+import { useAuth } from '../../hooks/useAuth'
+
+const Quiz = ({ timerValue, level }) => {
+  const apiEndpoint = "http://localhost:8000";
+
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [quizData, setQuizData] = useState({});
+  const [haveQuestions, setHaveQuestions] = useState(false);
+  const [isFinished, setIsFinished] = useState(false);
+  const [score, setScore] = useState(0);
+  const [btnDisabled, setBtnDisabled] = useState(false);
+  const [timer, setTimer] = useState(Date.now() + timerValue * 1000);
+  const [timerIndex, setTimerIndex] = useState(0);
+  const { user } = useAuth()
+  const countdownTimer = useRef();
+
+  useEffect(() => {
+    (async () => {
+      if (!haveQuestions) {
+        setQuestions(await getQuestions());
+        setHaveQuestions(true);
+      }
+    })();
+  }, [haveQuestions]);
+
+  useEffect(() => {
+    if (questions) {
+      setTimer(Date.now() + timerValue * 1000);
+    }
+  }, [questions]);
+
+  useEffect(() => {
+    if (isFinished) {
+      //updateStatistics();
+    }
+  }, [isFinished]);
 
-import './Quiz.css'
+  const getQuestions = async () => {
+    let numQuestions = 0;
 
-import ActiveQuiz from './ActiveQuiz'
-import FinalResult from '../FinalResult/FinalResult'
+    if (level === "easy") numQuestions = 1;
+    else if (level === "medium") numQuestions = 3;
+    else if (level === "hard") numQuestions = 5;
 
-import { ReactComponent as StopwatchIcon } from '../../assets/stopwatch-solid.svg'
+    const response = await fetch(apiEndpoint + `/questions?size=${numQuestions}`);
 
```

Comment:
Also, we are not taking in mind current language of the user, I think we must add it to the URL!!

Reply:
Yes, in principle it shouldn't be very complicated, I tried adding the lang parameter in the request but for some reason it didn't work, I left it pending.

Reply:
I think it is no working for you because the API key for question translation is not inserted.
You can find it on the GH Secrets section!
![image](https://github.com/Arquisoft/wiq_en3b/assets/109146445/908f30ad-3345-4ebf-bc51-4875e00cc71c)

Perhaps, you can try it out locally. Try to remove "***" for the Key found before. This part is in
questionService > src > services > translation-service.ts
![image](https://github.com/Arquisoft/wiq_en3b/assets/109146445/8b96e472-c2bc-4d35-a072-29f2fa6101fe)

Tell me if it worked! And if the error persist, post an Issue and the team we'll have a look